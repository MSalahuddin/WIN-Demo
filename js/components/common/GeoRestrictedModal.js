import React, { useContext, useEffect, useState } from 'react';
import { Linking, Platform } from 'react-native';
import PropTypes from 'prop-types';
import styled from 'styled-components/native';
import StampPopUp from './StampPopUp';
import TextButton from './TextButton'
import Text, { SIZE, FONT_FAMILY } from './Text';
import { UserContext } from '../../context/User.context';
import { BackgroundMusicContext } from '../../context/BackgroundMusic.context';
import { iconFailure, howtoBackgroundPopUp } from '../../../assets/images';
import { popUpStrings } from '../../stringConstants';
import { SOUNDS } from '../../soundUtils';
import { URLS, ERROR_STATUS_CODE } from '../../constants';
import { scaleHeight } from '../../platformUtils';
import { color } from '../../styles';
import Button, { BUTTON_COLOR_SCHEME } from './Button';
import SimpleButton from './SimpleButton';

const PopUpContentView = styled.View`
  align-items: center;
  justify-content: center;
`;

const TextWrapper = styled(Text)`
  margin-top: ${({ marginTop }) => (marginTop ? scaleHeight(marginTop) : 0)};
  margin-bottom: ${({ marginBottom }) => (marginBottom ? scaleHeight(marginBottom) : 0)};
`;


const ButtonContentContainer = styled.View`
  align-items: center;
  flex-direction: row;
  justify-content: center;
`;

const ButtonText = styled(Text)`
  margin-top: ${Platform.OS === 'ios' ? scaleHeight(5) : 0};
`;

const ButtonWrapper = styled(SimpleButton)`
`;

const TextButtonWrapper = styled(TextButton)`
  margin-top: ${scaleHeight(10)};
  margin-bottom: ${scaleHeight(10)};
`;

const GeoRestrictedModal = ({ children }) => {
  const { geoRestrictedStatusCode } = useContext(UserContext);
  const { playSoundEffect } = useContext(BackgroundMusicContext);
  const [isVisible, setIsVisible] = useState(false);

  useEffect(() => {
    if (
      geoRestrictedStatusCode === ERROR_STATUS_CODE.USER_LOCATION_RESTRICTED ||
      geoRestrictedStatusCode === ERROR_STATUS_CODE.USER_LOCATION_UNDETERMINED
    ) {
      setIsVisible(true);
      playSoundEffect(SOUNDS.POSITIVE_POPUP);
    }
    // eslint-disable-next-line react-hooks/exhaustive-deps
  }, [geoRestrictedStatusCode]);

  let popupBody = popUpStrings.regionNotSupported;
  let popupBannerTitle = popUpStrings.unsupportedRegion;

  if (geoRestrictedStatusCode === ERROR_STATUS_CODE.USER_LOCATION_UNDETERMINED) {
    popupBody = popUpStrings.locationCannotIdentified;
    popupBannerTitle = popUpStrings.locationError;
  }

  const renderPopupContent = () => {
    return (
      <PopUpContentView>
        <TextWrapper
          fontFamily={FONT_FAMILY.BOLD}
          size={SIZE.XXLARGE}
          color={color.white}
          alignCenter
        >
          {popupBannerTitle}
        </TextWrapper>
        <TextWrapper
          marginTop={20}
          marginBottom={20}
          fontFamily={FONT_FAMILY.MEDIUM}
          size={SIZE.SMALL}
          color={color.white}
          alignCenter
          
        >
          {popupBody}
        </TextWrapper>
        <TextWrapper
          fontFamily={FONT_FAMILY.LIGHT}
          size={SIZE.SMALL}
          color={color.white}
          alignCenter
        >
          {popUpStrings.pleaseContactSupportForDetails}
        </TextWrapper>
        <ButtonWrapper
          testID="popup-button"
          borderRadius={44}
          marginTop={25}
          height={53}
          width={200}
          onPress={() => {
            Linking.openURL(URLS.HELP);
          }}
          >
            <ButtonContentContainer>
                <ButtonText fontFamily={FONT_FAMILY.BOLD} color={color.darkNavyBlue} size={SIZE.LARGE}>
                   {popUpStrings.contactSupport}
                </ButtonText>
            </ButtonContentContainer>
          </ButtonWrapper>
      </PopUpContentView>
    );
  };

  return (
    <>
      {children}
      {/* <InstructionPopUp
        isVisible={true}
        mascot={sadChicken}
        bannerLabel={popupBannerTitle}
        buttonText={popUpStrings.contactSupport}
        icon={medalError}
        onPress={() => {
          // we are taking users out of the app
          // and freeze the app if their geo location
          // is not supported
          Linking.openURL(URLS.HELP);
        }}
        testID="geo-restricted-popup"
      >
        {renderPopupContent()}
      </InstructionPopUp> */}
      <StampPopUp
        isVisible={isVisible}
        backgroundImage={howtoBackgroundPopUp}
        stampIcon={iconFailure}
      >
        {renderPopupContent()}
      </StampPopUp>
    </>
  );
};

GeoRestrictedModal.propTypes = {
  children: PropTypes.node
};

GeoRestrictedModal.defaultProps = {
  children: null
};

export default GeoRestrictedModal;
