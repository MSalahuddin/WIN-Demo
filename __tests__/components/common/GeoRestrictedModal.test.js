import React from 'react';
import { Linking } from 'react-native';
import { render, fireEvent } from 'react-native-testing-library';
import sinon from 'sinon';

import GeoRestrictedModal from '../../../js/components/common/GeoRestrictedModal';
import MockProvider from '../../MockProvider';

describe('Component GeoRestrictedModal', () => {
  test('renders correctly', () => {
    const output = render(
      <MockProvider>
        <GeoRestrictedModal />
      </MockProvider>
    );
    expect(output).toMatchSnapshot();
  });

  test('renders correctly when geoRestrictedStatusCode = 451', () => {
    const output = render(
      <MockProvider geoRestrictedStatusCode={451}>
        <GeoRestrictedModal />
      </MockProvider>
    );
    expect(output).toMatchSnapshot();
  });

  test('renders correctly when geoRestrictedStatusCode = 551', () => {
    const output = render(
      <MockProvider geoRestrictedStatusCode={551}>
        <GeoRestrictedModal />
      </MockProvider>
    );
    expect(output).toMatchSnapshot();
  });

  test('calls Linking when geo-restricted-popup is pressed', () => {
    const openURLSpy = sinon.spy(Linking, 'openURL');
    const { getByTestId } = render(
      <MockProvider>
        <GeoRestrictedModal />
      </MockProvider>
    );
    fireEvent(getByTestId('geo-restricted-popup'), 'press');
    expect(openURLSpy).toHaveBeenCalled();
  });
});
