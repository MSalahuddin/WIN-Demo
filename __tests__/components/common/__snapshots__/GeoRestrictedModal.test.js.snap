// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component GeoRestrictedModal renders correctly 1`] = `
<RCTSafeAreaView
  emulateUnlessSupported={true}
>
  <Modal
    animationType="none"
    hardwareAccelerated={false}
    hideModalContentWhileAnimating={false}
    onModalHide={[Function]}
    onModalWillHide={[Function]}
    onModalWillShow={[Function]}
    onRequestClose={[Function]}
    scrollHorizontal={false}
    scrollOffset={0}
    scrollOffsetMax={0}
    scrollTo={null}
    supportedOrientations={
      Array [
        "portrait",
        "landscape",
      ]
    }
    swipeThreshold={100}
    transparent={true}
    visible={false}
  >
    <View
      accessible={true}
      focusable={true}
      forwardedRef={[Function]}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "backgroundColor": "black",
          "bottom": 0,
          "height": 1334,
          "left": 0,
          "opacity": 0,
          "position": "absolute",
          "right": 0,
          "top": 0,
          "width": 750,
        }
      }
    />
    <View
      forwardedRef={[Function]}
      hideModalContentWhileAnimating={false}
      onModalHide={[Function]}
      onModalWillHide={[Function]}
      onModalWillShow={[Function]}
      pointerEvents="box-none"
      scrollHorizontal={false}
      scrollOffset={0}
      scrollOffsetMax={0}
      scrollTo={null}
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "rgba(34,34,34, 0.6)",
          "flex": 1,
          "justifyContent": "flex-start",
          "margin": 37.5,
          "marginBottom": 0,
          "marginLeft": 0,
          "marginRight": 0,
          "marginTop": 0,
          "transform": Array [
            Object {
              "translateY": 0,
            },
          ],
        }
      }
      supportedOrientations={
        Array [
          "portrait",
          "landscape",
        ]
      }
      swipeThreshold={100}
    >
      <Image
        resizeMode="contain"
        source={1}
        style={
          Array [
            Object {
              "height": 176,
              "marginTop": 180,
              "width": 176,
            },
          ]
        }
      />
      <View
        isLowerCase={false}
        onLayout={[Function]}
        style={
          Array [
            Object {
              "alignItems": "center",
              "justifyContent": "center",
            },
            Object {
              "marginTop": 48,
            },
          ]
        }
        width={734}
      >
        <Image
          bannerType="NORMAL"
          resizeMode="contain"
          source={1}
          style={
            Array [
              Object {
                "height": 140.41739130434783,
                "width": 734,
              },
            ]
          }
          width={734}
        />
        <Text
          alignCenter={false}
          allowFontScaling={false}
          bannerType="NORMAL"
          color="#000000"
          fontFamily="BOLD"
          fontSize={22}
          isUnderlined={false}
          isUppercase={true}
          maxFontSizeMultiplier={1.3}
          size="BANNER_LARGE"
          style={
            Array [
              Object {
                "color": "#000000",
                "fontFamily": "CalibreBold",
                "fontSize": 34.5,
                "justifyContent": "center",
                "textAlign": "left",
                "textDecorationColor": "#000000",
                "textDecorationLine": "none",
                "textDecorationStyle": "solid",
                "textTransform": "uppercase",
              },
              Object {
                "color": "#FFFFFF",
                "letterSpacing": 1.5,
                "position": "absolute",
              },
            ]
          }
        >
          Unsupported region
        </Text>
      </View>
      <View
        hasChildren={true}
        style={
          Array [
            Object {
              "alignItems": "center",
              "backgroundColor": "#FFFFFF",
              "borderBottomLeftRadius": 16,
              "borderBottomRightRadius": 16,
              "justifyContent": "flex-start",
              "marginTop": -40,
              "paddingBottom": 120,
              "paddingTop": 80,
              "shadowColor": "rgba(53, 53, 53, 0.3)",
              "shadowOffset": Object {
                "height": 4,
                "width": 4,
              },
              "shadowOpacity": 1,
              "shadowRadius": 2,
              "width": "80%",
              "zIndex": -1,
            },
          ]
        }
      >
        <View
          style={
            Array [
              Object {
                "alignItems": "center",
                "justifyContent": "center",
                "paddingHorizontal": 22.5,
              },
            ]
          }
        >
          <Text
            alignCenter={true}
            allowFontScaling={true}
            color="#353535"
            fontFamily="MEDIUM"
            isUnderlined={false}
            isUppercase={false}
            maxFontSizeMultiplier={1.3}
            size="SMALL"
            style={
              Array [
                Object {
                  "color": "#353535",
                  "fontFamily": "Calibre-Medium",
                  "fontSize": 27,
                  "justifyContent": "center",
                  "textAlign": "center",
                  "textDecorationColor": "#353535",
                  "textDecorationLine": "none",
                  "textDecorationStyle": "solid",
                  "textTransform": "none",
                },
              ]
            }
          >
            Sorry, your current region does not allow you to play these games.
          </Text>
          <Text
            alignCenter={true}
            allowFontScaling={true}
            color="#353535"
            fontFamily="REGULAR"
            isUnderlined={false}
            isUppercase={false}
            marginTop={16}
            maxFontSizeMultiplier={1.3}
            size="SMALL"
            style={
              Array [
                Object {
                  "color": "#353535",
                  "fontFamily": "CalibreRegular",
                  "fontSize": 27,
                  "justifyContent": "center",
                  "textAlign": "center",
                  "textDecorationColor": "#353535",
                  "textDecorationLine": "none",
                  "textDecorationStyle": "solid",
                  "textTransform": "none",
                },
                Object {
                  "marginTop": 32,
                },
              ]
            }
          >
            Please contact support for more details
          </Text>
        </View>
      </View>
      <View
        onLayout={[Function]}
        style={
          Array [
            Object {
              "alignItems": "center",
              "justifyContent": "center",
              "marginHorizontal": 10,
              "marginVertical": 10,
            },
          ]
        }
      >
        <View
          accessible={true}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "alignItems": "center",
              "backgroundColor": "rgba(96,202,142,1)",
              "borderRadius": 88,
              "height": 176,
              "justifyContent": "center",
              "marginTop": -92,
              "opacity": 1,
              "width": 542,
            }
          }
          testID="popup-button"
        >
          <View
            borderRadius={44}
            colorScheme="rgba(128,229,173,1)"
            height={88}
            style={
              Array [
                Object {
                  "alignItems": "center",
                  "backgroundColor": "rgba(128,229,173,1)",
                  "borderRadius": 88,
                  "height": 166,
                  "justifyContent": "center",
                  "width": 536,
                },
              ]
            }
            width={271}
          >
            <View
              borderRadius={44}
              colorScheme="rgba(114,213,158,1)"
              height={88}
              style={
                Array [
                  Object {
                    "alignItems": "center",
                    "backgroundColor": "rgba(114,213,158,1)",
                    "borderRadius": 108,
                    "height": 162,
                    "justifyContent": "center",
                    "marginTop": 8,
                    "width": 536,
                  },
                ]
              }
              width={271}
            >
              <View
                style={
                  Array [
                    Object {
                      "alignItems": "center",
                      "flexDirection": "row",
                      "justifyContent": "center",
                    },
                  ]
                }
              >
                <Text
                  alignCenter={false}
                  allowFontScaling={true}
                  color="#FFFFFF"
                  fontFamily="MEDIUM"
                  isUnderlined={false}
                  isUppercase={false}
                  maxFontSizeMultiplier={1.3}
                  size="XLARGE"
                  style={
                    Array [
                      Object {
                        "color": "#FFFFFF",
                        "fontFamily": "Calibre-Medium",
                        "fontSize": 36,
                        "justifyContent": "center",
                        "textAlign": "left",
                        "textDecorationColor": "#FFFFFF",
                        "textDecorationLine": "none",
                        "textDecorationStyle": "solid",
                        "textTransform": "none",
                      },
                      Object {
                        "marginTop": 14,
                      },
                    ]
                  }
                >
                  Contact support
                </Text>
              </View>
            </View>
          </View>
        </View>
      </View>
      <Image
        resizeMode="contain"
        source={1}
        style={
          Array [
            Object {
              "height": 328,
              "position": "absolute",
              "right": 56,
              "top": -304,
              "width": 208,
            },
          ]
        }
        top={0}
      />
    </View>
  </Modal>
</RCTSafeAreaView>
`;

exports[`Component GeoRestrictedModal renders correctly when geoRestrictedStatusCode = 451 1`] = `
<RCTSafeAreaView
  emulateUnlessSupported={true}
>
  <Modal
    animationType="none"
    hardwareAccelerated={false}
    hideModalContentWhileAnimating={false}
    onModalHide={[Function]}
    onModalWillHide={[Function]}
    onModalWillShow={[Function]}
    onRequestClose={[Function]}
    scrollHorizontal={false}
    scrollOffset={0}
    scrollOffsetMax={0}
    scrollTo={null}
    supportedOrientations={
      Array [
        "portrait",
        "landscape",
      ]
    }
    swipeThreshold={100}
    transparent={true}
    visible={false}
  >
    <View
      accessible={true}
      focusable={true}
      forwardedRef={[Function]}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "backgroundColor": "black",
          "bottom": 0,
          "height": 1334,
          "left": 0,
          "opacity": 0,
          "position": "absolute",
          "right": 0,
          "top": 0,
          "width": 750,
        }
      }
    />
    <View
      forwardedRef={[Function]}
      hideModalContentWhileAnimating={false}
      onModalHide={[Function]}
      onModalWillHide={[Function]}
      onModalWillShow={[Function]}
      pointerEvents="box-none"
      scrollHorizontal={false}
      scrollOffset={0}
      scrollOffsetMax={0}
      scrollTo={null}
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "rgba(34,34,34, 0.6)",
          "flex": 1,
          "justifyContent": "flex-start",
          "margin": 37.5,
          "marginBottom": 0,
          "marginLeft": 0,
          "marginRight": 0,
          "marginTop": 0,
          "transform": Array [
            Object {
              "translateY": 0,
            },
          ],
        }
      }
      supportedOrientations={
        Array [
          "portrait",
          "landscape",
        ]
      }
      swipeThreshold={100}
    >
      <Image
        resizeMode="contain"
        source={1}
        style={
          Array [
            Object {
              "height": 176,
              "marginTop": 180,
              "width": 176,
            },
          ]
        }
      />
      <View
        isLowerCase={false}
        onLayout={[Function]}
        style={
          Array [
            Object {
              "alignItems": "center",
              "justifyContent": "center",
            },
            Object {
              "marginTop": 48,
            },
          ]
        }
        width={734}
      >
        <Image
          bannerType="NORMAL"
          resizeMode="contain"
          source={1}
          style={
            Array [
              Object {
                "height": 140.41739130434783,
                "width": 734,
              },
            ]
          }
          width={734}
        />
        <Text
          alignCenter={false}
          allowFontScaling={false}
          bannerType="NORMAL"
          color="#000000"
          fontFamily="BOLD"
          fontSize={22}
          isUnderlined={false}
          isUppercase={true}
          maxFontSizeMultiplier={1.3}
          size="BANNER_LARGE"
          style={
            Array [
              Object {
                "color": "#000000",
                "fontFamily": "CalibreBold",
                "fontSize": 34.5,
                "justifyContent": "center",
                "textAlign": "left",
                "textDecorationColor": "#000000",
                "textDecorationLine": "none",
                "textDecorationStyle": "solid",
                "textTransform": "uppercase",
              },
              Object {
                "color": "#FFFFFF",
                "letterSpacing": 1.5,
                "position": "absolute",
              },
            ]
          }
        >
          Unsupported region
        </Text>
      </View>
      <View
        hasChildren={true}
        style={
          Array [
            Object {
              "alignItems": "center",
              "backgroundColor": "#FFFFFF",
              "borderBottomLeftRadius": 16,
              "borderBottomRightRadius": 16,
              "justifyContent": "flex-start",
              "marginTop": -40,
              "paddingBottom": 120,
              "paddingTop": 80,
              "shadowColor": "rgba(53, 53, 53, 0.3)",
              "shadowOffset": Object {
                "height": 4,
                "width": 4,
              },
              "shadowOpacity": 1,
              "shadowRadius": 2,
              "width": "80%",
              "zIndex": -1,
            },
          ]
        }
      >
        <View
          style={
            Array [
              Object {
                "alignItems": "center",
                "justifyContent": "center",
                "paddingHorizontal": 22.5,
              },
            ]
          }
        >
          <Text
            alignCenter={true}
            allowFontScaling={true}
            color="#353535"
            fontFamily="MEDIUM"
            isUnderlined={false}
            isUppercase={false}
            maxFontSizeMultiplier={1.3}
            size="SMALL"
            style={
              Array [
                Object {
                  "color": "#353535",
                  "fontFamily": "Calibre-Medium",
                  "fontSize": 27,
                  "justifyContent": "center",
                  "textAlign": "center",
                  "textDecorationColor": "#353535",
                  "textDecorationLine": "none",
                  "textDecorationStyle": "solid",
                  "textTransform": "none",
                },
              ]
            }
          >
            Sorry, your current region does not allow you to play these games.
          </Text>
          <Text
            alignCenter={true}
            allowFontScaling={true}
            color="#353535"
            fontFamily="REGULAR"
            isUnderlined={false}
            isUppercase={false}
            marginTop={16}
            maxFontSizeMultiplier={1.3}
            size="SMALL"
            style={
              Array [
                Object {
                  "color": "#353535",
                  "fontFamily": "CalibreRegular",
                  "fontSize": 27,
                  "justifyContent": "center",
                  "textAlign": "center",
                  "textDecorationColor": "#353535",
                  "textDecorationLine": "none",
                  "textDecorationStyle": "solid",
                  "textTransform": "none",
                },
                Object {
                  "marginTop": 32,
                },
              ]
            }
          >
            Please contact support for more details
          </Text>
        </View>
      </View>
      <View
        onLayout={[Function]}
        style={
          Array [
            Object {
              "alignItems": "center",
              "justifyContent": "center",
              "marginHorizontal": 10,
              "marginVertical": 10,
            },
          ]
        }
      >
        <View
          accessible={true}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "alignItems": "center",
              "backgroundColor": "rgba(96,202,142,1)",
              "borderRadius": 88,
              "height": 176,
              "justifyContent": "center",
              "marginTop": -92,
              "opacity": 1,
              "width": 542,
            }
          }
          testID="popup-button"
        >
          <View
            borderRadius={44}
            colorScheme="rgba(128,229,173,1)"
            height={88}
            style={
              Array [
                Object {
                  "alignItems": "center",
                  "backgroundColor": "rgba(128,229,173,1)",
                  "borderRadius": 88,
                  "height": 166,
                  "justifyContent": "center",
                  "width": 536,
                },
              ]
            }
            width={271}
          >
            <View
              borderRadius={44}
              colorScheme="rgba(114,213,158,1)"
              height={88}
              style={
                Array [
                  Object {
                    "alignItems": "center",
                    "backgroundColor": "rgba(114,213,158,1)",
                    "borderRadius": 108,
                    "height": 162,
                    "justifyContent": "center",
                    "marginTop": 8,
                    "width": 536,
                  },
                ]
              }
              width={271}
            >
              <View
                style={
                  Array [
                    Object {
                      "alignItems": "center",
                      "flexDirection": "row",
                      "justifyContent": "center",
                    },
                  ]
                }
              >
                <Text
                  alignCenter={false}
                  allowFontScaling={true}
                  color="#FFFFFF"
                  fontFamily="MEDIUM"
                  isUnderlined={false}
                  isUppercase={false}
                  maxFontSizeMultiplier={1.3}
                  size="XLARGE"
                  style={
                    Array [
                      Object {
                        "color": "#FFFFFF",
                        "fontFamily": "Calibre-Medium",
                        "fontSize": 36,
                        "justifyContent": "center",
                        "textAlign": "left",
                        "textDecorationColor": "#FFFFFF",
                        "textDecorationLine": "none",
                        "textDecorationStyle": "solid",
                        "textTransform": "none",
                      },
                      Object {
                        "marginTop": 14,
                      },
                    ]
                  }
                >
                  Contact support
                </Text>
              </View>
            </View>
          </View>
        </View>
      </View>
      <Image
        resizeMode="contain"
        source={1}
        style={
          Array [
            Object {
              "height": 328,
              "position": "absolute",
              "right": 56,
              "top": -304,
              "width": 208,
            },
          ]
        }
        top={0}
      />
    </View>
  </Modal>
</RCTSafeAreaView>
`;

exports[`Component GeoRestrictedModal renders correctly when geoRestrictedStatusCode = 551 1`] = `
<RCTSafeAreaView
  emulateUnlessSupported={true}
>
  <Modal
    animationType="none"
    hardwareAccelerated={false}
    hideModalContentWhileAnimating={false}
    onModalHide={[Function]}
    onModalWillHide={[Function]}
    onModalWillShow={[Function]}
    onRequestClose={[Function]}
    scrollHorizontal={false}
    scrollOffset={0}
    scrollOffsetMax={0}
    scrollTo={null}
    supportedOrientations={
      Array [
        "portrait",
        "landscape",
      ]
    }
    swipeThreshold={100}
    transparent={true}
    visible={false}
  >
    <View
      accessible={true}
      focusable={true}
      forwardedRef={[Function]}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "backgroundColor": "black",
          "bottom": 0,
          "height": 1334,
          "left": 0,
          "opacity": 0,
          "position": "absolute",
          "right": 0,
          "top": 0,
          "width": 750,
        }
      }
    />
    <View
      forwardedRef={[Function]}
      hideModalContentWhileAnimating={false}
      onModalHide={[Function]}
      onModalWillHide={[Function]}
      onModalWillShow={[Function]}
      pointerEvents="box-none"
      scrollHorizontal={false}
      scrollOffset={0}
      scrollOffsetMax={0}
      scrollTo={null}
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "rgba(34,34,34, 0.6)",
          "flex": 1,
          "justifyContent": "flex-start",
          "margin": 37.5,
          "marginBottom": 0,
          "marginLeft": 0,
          "marginRight": 0,
          "marginTop": 0,
          "transform": Array [
            Object {
              "translateY": 0,
            },
          ],
        }
      }
      supportedOrientations={
        Array [
          "portrait",
          "landscape",
        ]
      }
      swipeThreshold={100}
    >
      <Image
        resizeMode="contain"
        source={1}
        style={
          Array [
            Object {
              "height": 176,
              "marginTop": 180,
              "width": 176,
            },
          ]
        }
      />
      <View
        isLowerCase={false}
        onLayout={[Function]}
        style={
          Array [
            Object {
              "alignItems": "center",
              "justifyContent": "center",
            },
            Object {
              "marginTop": 48,
            },
          ]
        }
        width={734}
      >
        <Image
          bannerType="NORMAL"
          resizeMode="contain"
          source={1}
          style={
            Array [
              Object {
                "height": 140.41739130434783,
                "width": 734,
              },
            ]
          }
          width={734}
        />
        <Text
          alignCenter={false}
          allowFontScaling={false}
          bannerType="NORMAL"
          color="#000000"
          fontFamily="BOLD"
          fontSize={22}
          isUnderlined={false}
          isUppercase={true}
          maxFontSizeMultiplier={1.3}
          size="BANNER_LARGE"
          style={
            Array [
              Object {
                "color": "#000000",
                "fontFamily": "CalibreBold",
                "fontSize": 34.5,
                "justifyContent": "center",
                "textAlign": "left",
                "textDecorationColor": "#000000",
                "textDecorationLine": "none",
                "textDecorationStyle": "solid",
                "textTransform": "uppercase",
              },
              Object {
                "color": "#FFFFFF",
                "letterSpacing": 1.5,
                "position": "absolute",
              },
            ]
          }
        >
          Unsupported region
        </Text>
      </View>
      <View
        hasChildren={true}
        style={
          Array [
            Object {
              "alignItems": "center",
              "backgroundColor": "#FFFFFF",
              "borderBottomLeftRadius": 16,
              "borderBottomRightRadius": 16,
              "justifyContent": "flex-start",
              "marginTop": -40,
              "paddingBottom": 120,
              "paddingTop": 80,
              "shadowColor": "rgba(53, 53, 53, 0.3)",
              "shadowOffset": Object {
                "height": 4,
                "width": 4,
              },
              "shadowOpacity": 1,
              "shadowRadius": 2,
              "width": "80%",
              "zIndex": -1,
            },
          ]
        }
      >
        <View
          style={
            Array [
              Object {
                "alignItems": "center",
                "justifyContent": "center",
                "paddingHorizontal": 22.5,
              },
            ]
          }
        >
          <Text
            alignCenter={true}
            allowFontScaling={true}
            color="#353535"
            fontFamily="MEDIUM"
            isUnderlined={false}
            isUppercase={false}
            maxFontSizeMultiplier={1.3}
            size="SMALL"
            style={
              Array [
                Object {
                  "color": "#353535",
                  "fontFamily": "Calibre-Medium",
                  "fontSize": 27,
                  "justifyContent": "center",
                  "textAlign": "center",
                  "textDecorationColor": "#353535",
                  "textDecorationLine": "none",
                  "textDecorationStyle": "solid",
                  "textTransform": "none",
                },
              ]
            }
          >
            Sorry, your current region does not allow you to play these games.
          </Text>
          <Text
            alignCenter={true}
            allowFontScaling={true}
            color="#353535"
            fontFamily="REGULAR"
            isUnderlined={false}
            isUppercase={false}
            marginTop={16}
            maxFontSizeMultiplier={1.3}
            size="SMALL"
            style={
              Array [
                Object {
                  "color": "#353535",
                  "fontFamily": "CalibreRegular",
                  "fontSize": 27,
                  "justifyContent": "center",
                  "textAlign": "center",
                  "textDecorationColor": "#353535",
                  "textDecorationLine": "none",
                  "textDecorationStyle": "solid",
                  "textTransform": "none",
                },
                Object {
                  "marginTop": 32,
                },
              ]
            }
          >
            Please contact support for more details
          </Text>
        </View>
      </View>
      <View
        onLayout={[Function]}
        style={
          Array [
            Object {
              "alignItems": "center",
              "justifyContent": "center",
              "marginHorizontal": 10,
              "marginVertical": 10,
            },
          ]
        }
      >
        <View
          accessible={true}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "alignItems": "center",
              "backgroundColor": "rgba(96,202,142,1)",
              "borderRadius": 88,
              "height": 176,
              "justifyContent": "center",
              "marginTop": -92,
              "opacity": 1,
              "width": 542,
            }
          }
          testID="popup-button"
        >
          <View
            borderRadius={44}
            colorScheme="rgba(128,229,173,1)"
            height={88}
            style={
              Array [
                Object {
                  "alignItems": "center",
                  "backgroundColor": "rgba(128,229,173,1)",
                  "borderRadius": 88,
                  "height": 166,
                  "justifyContent": "center",
                  "width": 536,
                },
              ]
            }
            width={271}
          >
            <View
              borderRadius={44}
              colorScheme="rgba(114,213,158,1)"
              height={88}
              style={
                Array [
                  Object {
                    "alignItems": "center",
                    "backgroundColor": "rgba(114,213,158,1)",
                    "borderRadius": 108,
                    "height": 162,
                    "justifyContent": "center",
                    "marginTop": 8,
                    "width": 536,
                  },
                ]
              }
              width={271}
            >
              <View
                style={
                  Array [
                    Object {
                      "alignItems": "center",
                      "flexDirection": "row",
                      "justifyContent": "center",
                    },
                  ]
                }
              >
                <Text
                  alignCenter={false}
                  allowFontScaling={true}
                  color="#FFFFFF"
                  fontFamily="MEDIUM"
                  isUnderlined={false}
                  isUppercase={false}
                  maxFontSizeMultiplier={1.3}
                  size="XLARGE"
                  style={
                    Array [
                      Object {
                        "color": "#FFFFFF",
                        "fontFamily": "Calibre-Medium",
                        "fontSize": 36,
                        "justifyContent": "center",
                        "textAlign": "left",
                        "textDecorationColor": "#FFFFFF",
                        "textDecorationLine": "none",
                        "textDecorationStyle": "solid",
                        "textTransform": "none",
                      },
                      Object {
                        "marginTop": 14,
                      },
                    ]
                  }
                >
                  Contact support
                </Text>
              </View>
            </View>
          </View>
        </View>
      </View>
      <Image
        resizeMode="contain"
        source={1}
        style={
          Array [
            Object {
              "height": 328,
              "position": "absolute",
              "right": 56,
              "top": -304,
              "width": 208,
            },
          ]
        }
        top={0}
      />
    </View>
  </Modal>
</RCTSafeAreaView>
`;
